bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -993,56 +1006,58 @@ synchronized boolean readEntryLogHardLimit(long size) {
 
     public ByteBuf internalReadEntry(long ledgerId, long entryId, long location)
             throws IOException, Bookie.NoEntryException {
-        long entryLogId = logIdForOffset(location);
-        long pos = location & 0xffffffffL;
-        ByteBuf sizeBuff = sizeBuffer.get();
-        sizeBuff.clear();
-        pos -= 4; // we want to get the ledgerId and length to check
-        BufferedReadChannel fc;
+        EntryLogBufferedReadChannel bc = null;
         try {
-            fc = getChannelForLogId(entryLogId);
+            long entryLogId = logIdForOffset(location);
```

Comment:
What is this change for? It looks pretty much the same as it was before, with the exception handling changed.

Reply:
If you're going to change around the exception handling, you may as well make EntryLogBufferedReadChannel autocloseable, and use a try-with-resources block.